### ISSUE TITLE
A restart after a sync can cause the node to panic due to a missing bundle instance

### ISSUE THREAD

**Issue by luca-moser** [iotaledger/hornet#53]
**Describe the bug**
After a sync and then a restart, the node paniced with not being able to find a bundle corresponding to a transaction it wanted to confirm:
```
2019/12/17 11:51:01 [ PANIC ] Tangle: confirmMilestone: Tx: XUNCAVRNAOHNJZID9SJ9XTGCTTQ9KPDTBBWHEMKWOCCNYK9RACCUZJTOEHDZDFZWUJLWKXCAZYZH99999, Bundle not found: VAJOHANFEOTRSIPCLG9MIPENDFPLQQUGSBLBHMKZ9XVCUSWIKJOOHSPWJAXVLPTAKMPURYAYD9ONODVOW
panic: [ PANIC ] Tangle: confirmMilestone: Tx: XUNCAVRNAOHNJZID9SJ9XTGCTTQ9KPDTBBWHEMKWOCCNYK9RACCUZJTOEHDZDFZWUJLWKXCAZYZH99999, Bundle not found: VAJOHANFEOTRSIPCLG9MIPENDFPLQQUGSBLBHMKZ9XVCUSWIKJOOHSPWJAXVLPTAKMPURYAYD9ONODVOW

goroutine 404241 [running]:
log.(*Logger).Panicln(0xc000144000, 0xc0083f78a8, 0x2, 0x2)
        /usr/local/go/src/log/log.go:226 +0xaa
github.com/gohornet/hornet/packages/logger.(*Logger).Panicf(0xc0002c53e0, 0xe983ce, 0x2e, 0xc0083f7a20, 0x2, 0x2)
        /__w/hornet/hornet/packages/logger/logger.go:100 +0x1cc
github.com/gohornet/hornet/plugins/tangle.confirmMilestone(0xc00013882b, 0xc0440a52d0)
        /__w/hornet/hornet/plugins/tangle/confirmation.go:73 +0x7ab
github.com/gohornet/hornet/plugins/tangle.solidifyMilestone(0xc000000000)
        /__w/hornet/hornet/plugins/tangle/solidifier.go:334 +0x5e6
github.com/gohornet/hornet/plugins/tangle.configureTangleProcessor.func3(0xc0433da100, 0x1, 0x1, 0xc04b3e7260)
        /__w/hornet/hornet/plugins/tangle/tangle_processor.go:52 +0x4d
github.com/gohornet/hornet/packages/workerpool.(*WorkerPool).startWorkers.func1(0xc000144c30)
        /__w/hornet/hornet/packages/workerpool/workerpool.go:159 +0x1ad
created by github.com/gohornet/hornet/packages/workerpool.(*WorkerPool).startWorkers
        /__w/hornet/hornet/packages/workerpool/workerpool.go:135 +0x5e
```

Theoretically this can happen because we're triggering a confirmation process for a given milestone, while we didn't yet complete the assignment of a transaction to its bundle.
Note that we:
1. store a transaction in the LRU cache
2. the submit an address persister task
3. only then map the transaction to its bundle bucket

**To Reproduce**
- Sync up from LS
- Restart
- Maybe panics

**Expected behavior**
Shouldn't panic.

**Environment information:**
 - OS: Ubuntu 16.04
 - RAM: 64 GB
 - Cores: 8 Cores
 - Type: Hetzner dedicated server
 - HORNET version  0.2.7

**Comment by centercirclesolutions** [comment]
This still happened after Merge pull request #62 

I updated the local source and built with tag pow_avx. then stopped Hornet and restarted. This error occurred shortly after restart


**Environment information:**

- OS: Ubuntu Server 19.04
- RAM: 8 GB
- Cores: 4 Cores
- Type: HP Intel i5 mini PC
- HORNET version 0.2.7 + fixes

```2019/12/17 11:50:32 [ INFO ] Tangle: Run solidity check for Milestone (1280459)...
2019/12/17 11:50:32 [ INFO ] Tangle: Solidifier finished (9): passed: true, tx: 13, collect: 1.50931ms, total: 1.575247ms, entryTx: 0
2019/12/17 11:50:32 [ PANIC ] Tangle: confirmMilestone: Tx: YXVJAVJHCCWNSYETQHBYO9UZAOHNKIOTAJ9KOOFYYWIUKIDPTIAXQIPLOICV9LNSGXQJKRBKHGGLA9999, Bundle not valid: VAJOHANFEOTRSIPCLG9MIPENDFPLQQUGSBLBHMKZ9XVCUSWIKJOOHSPWJAXVLPTAKMPURYAYD9ON
ODVOW
panic: [ PANIC ] Tangle: confirmMilestone: Tx: YXVJAVJHCCWNSYETQHBYO9UZAOHNKIOTAJ9KOOFYYWIUKIDPTIAXQIPLOICV9LNSGXQJKRBKHGGLA9999, Bundle not valid: VAJOHANFEOTRSIPCLG9MIPENDFPLQQUGSBLBHMKZ9XVCUSWIKJOOHSPWJAXVLPTAKMPURYAYD9ONODVOW
goroutine 373884 [running]:
log.(*Logger).Panicln(0xc000124000, 0xc01963b8a8, 0x2, 0x2)
        /usr/local/go/src/log/log.go:226 +0xaa
github.com/gohornet/hornet/packages/logger.(*Logger).Panicf(0xc0002a2c00, 0xe9e376, 0x2e, 0xc01963ba20, 0x2, 0x2)
        /home/plex/hornet/packages/logger/logger.go:100 +0x1cc
github.com/gohornet/hornet/plugins/tangle.confirmMilestone(0xc0001389cb, 0xc00015ca10)
        /home/plex/hornet/plugins/tangle/confirmation.go:77 +0x6d2
github.com/gohornet/hornet/plugins/tangle.solidifyMilestone(0xc000000000)
        /home/plex/hornet/plugins/tangle/solidifier.go:334 +0x5e6
github.com/gohornet/hornet/plugins/tangle.configureTangleProcessor.func3(0xc04d230b20, 0x1, 0x1, 0xc02f0485a0)
        /home/plex/hornet/plugins/tangle/tangle_processor.go:52 +0x4d
github.com/gohornet/hornet/packages/workerpool.(*WorkerPool).startWorkers.func1(0xc000332370)
        /home/plex/hornet/packages/workerpool/workerpool.go:159 +0x1ad
created by github.com/gohornet/hornet/packages/workerpool.(*WorkerPool).startWorkers
        /home/plex/hornet/packages/workerpool/workerpool.go:135 +0x5e
plex@plexserver:~/hornet$```

**Comment by muXxer** [comment]
Should be solved with #199
